atc/creds/vault/vault_test.go

```diff
@@ -274,3 +297,687 @@ var _ = Describe("Vault", func() {
 		})
 	})
 })
+
+// The below tests use ghttp handlers to mock a real vault API to the api_client.
+// AppendHandlers has the following behavior which make the tests a bit messy.
```

Comment:
Just FYI, an alternative to `AppendHandlers` in situations where you don't care about the order is [`RouteToHandler`](https://pkg.go.dev/github.com/onsi/gomega/ghttp#Server.RouteToHandler).

Reply:
Thanks! This is what I was looking for and apparently totally missed. Used it to clean up the mount calls that get made everywhere but left the others to show the proper flow of expected calls.